### PR TITLE
Switched mask from uint8 to bool in concat layers

### PR DESCRIPTION
**Pull request by matt-gardner**
Addresses #5867.  It looks like everywhere else in the code, masks are of type `bool`.  The cast to `uint8` causes a problem when one of the concated inputs has a mask and one doesn't, because you can't concat a `bool` tensor and a `uint8` tensor.

### CODE REVIEW

**Comment by matt-gardner** [comment]
Looking into the test failure. Will fix.

**Comment by matt-gardner** [comment]
@fchollet, looks like that fixed the test.  Let me know if you think standardizing the dtype of masks should be done another way.

File: keras/layers/core.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def __init__(self, mask_value=0., **kwargs):
         self.mask_value = mask_value
 
     def compute_mask(self, inputs, mask=None):
-        return K.any(K.not_equal(inputs, self.mask_value), axis=-1)
+        return K.cast(K.any(K.not_equal(inputs, self.mask_value), axis=-1), 'bool')
```
Comment (fchollet) [inline keras/layers/core.py]: Are there cases where `any` would not return `bool`? If so, better to fix it in the corresponding backend.

Comment (matt-gardner) [inline keras/layers/core.py]: [`K.any()`](https://github.com/fchollet/keras/blob/bd9214a5470ace245a6915d4e6d448401360e5cf/keras/backend/tensorflow_backend.py#L1140) and [`K.all()`](https://github.com/fchollet/keras/blob/bd9214a5470ace245a6915d4e6d448401360e5cf/keras/backend/tensorflow_backend.py#L1157) in the tf backend explicitly cast to `uint8`.  I'll change them there.

Comment (matt-gardner) [inline keras/layers/core.py]: Though, the issue with standardizing to `bool` instead of `uint8` is that you have to cast before a multiplication.  You can't do `K.any(x) * K.any(y)`.  Should I switch things to `uint8` instead?

**Comment by fchollet** [comment]
LGTM
